tests/Feature/Api/LoginTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Api;
+
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Laravel\Sanctum\Sanctum;
+use Symfony\Component\HttpFoundation\Response;
+use Tests\TestCase;
+
+class LoginTest extends TestCase
+{
+    public function testUserCanLoginWithValidCredentials(): void
+    {
+        User::factory()->create([
+            "email" => "email@example.com",
+            "password" => Hash::make("password@example"),
+        ]);
+
+        $response = $this->postJson("/api/login", [
+            "email" => "email@example.com",
+            "password" => "password@example",
+        ]);
+
+        $response->assertJsonStructure([
+            "access_token",
+        ]);
+    }
+
+    public function testAuthenticatedUserCanLogout(): void
+    {
+        Sanctum::actingAs(User::factory()->create());
+
+        $response = $this->postJson("/api/logout");
+
+        $response->assertJson([
+            "message" => "Logged out.",
+        ]);
+    }
+
+    public function testUserCannotLogInWithInvalidCredentials(): void
```

Comment:
I think it will be better in this way -> one test = one assertStatus. So here we should have two tests. If tests failed we will know exactly what test failed.

Reply:
I sometimes add assertions i middle of test, e.g. while testing if able to delete, I'll assert first it's created, should i remove those?